src/repo.rs

```diff
@@ -2100,6 +2101,40 @@ impl Repository {
         }
         Ok(ignored == 1)
     }
+
+    /// Look up the value of one git attribute for path. Equivalent to libgit2::git_attr_get.
+    pub fn get_attr(&self, flags: CheckAttributeFlags, path: &str, name: &str) -> Result<AttributeType, Error> {
+        let mut value_out : *mut c_char = 0 as *mut c_char;
+        let value_out_ptr : *mut *mut c_char = &mut value_out; 
+        
+        let c_flags = flags.bits() as c_uint;
+        let c_path = CString::new(path).expect("The path value contained an invalid 0 byte within the string");
+        let c_name = CString::new(name).expect("The name value contained an invalid 0 byte within the string");
+
+        let attribute_type = unsafe {
+            try_call!(raw::git_attr_get(value_out_ptr, self.raw, c_flags, c_path, c_name));
+
+            let attr_type = raw::git_attr_value(*value_out_ptr);
+
+            let attribute_type = match attr_type {
+                raw::GIT_ATTR_UNSPECIFIED => AttributeType::Unspecified,
+                raw::GIT_ATTR_TRUE => AttributeType::True,
+                raw::GIT_ATTR_FALSE => AttributeType::False,
+                raw::GIT_ATTR_VALUE => {
+                    let value = CStr::from_ptr(*value_out_ptr);
+                    let string_value = value.to_string_lossy().to_owned().to_string();
+                    AttributeType::Value(string_value)
```

Comment:
For the returned attribute string value, I've just copied it out lossily to a rust string. It might be possible to hold a borrow, but that seems like a pretty big pain, and I'm not entirely sure how long that memory is valid anyhow.